pulsar-client-cpp/lib/ConsumerImpl.cc

```diff
@@ -923,4 +924,75 @@ void ConsumerImpl::seekAsync(const MessageId& msgId, ResultCallback callback) {
 
 bool ConsumerImpl::isReadCompacted() { return readCompacted_; }
 
+void ConsumerImpl::hasMessageAvailableAsync(HasMessageAvailableCallback callback) {
+    MessageId lastDequed = this->lastMessageIdDequed();
+    MessageId lastInBroker = this->lastMessageIdInBroker();
+    if (lastInBroker > lastDequed && lastInBroker.entryId() != -1) {
```

Comment:
What about the case where the broker a lot of messages, but has just recently rolled the ledger?

Reply:
Thanks, Yes, There was a PR fix the case. Before that PR, the lastIDInBroker is the start of the new ledger, After that fix, the lastIDInBroker is the end of former ledger.